cmd/segment-reaper/observer.go

```diff
@@ -175,13 +175,37 @@ func (obsvr *observer) analyzeProject(ctx context.Context) error {
 			}
 
 			brokenObject := false
+			// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
 			if object.hasLastSegment {
 				if object.expectedNumberOfSegments == 0 {
 					if !object.segments.IsSequence() {
 						brokenObject = true
 					}
-				} else if object.segments.Count() != int(object.expectedNumberOfSegments)-1 {
-					// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
+				} else if object.segments.Count() < int(object.expectedNumberOfSegments)-1 {
+					brokenObject = true
+				} else if object.segments.Count() > int(object.expectedNumberOfSegments)-1 {
+					// verify if initial sequence is valid
+					for index := 0; index < int(object.expectedNumberOfSegments)-1; index++ {
```

Comment:
nit: We could have a method in bitwise to verify a subsequence or modify the `IsSequence` to pass an int which indicates until what index and pass `< 0` for indicating to verify all.

Reply:
agree but I would prefer to focus on testing and maybe later think about cleaning code